lib/Controller/SettingsController.php

```diff
@@ -310,4 +312,31 @@ private function setSlotDuration(string $value):JSONResponse {
 
 		return new JSONResponse();
 	}
+
+	/**
+	 * sets defaultReminder for user
+	 *
+	 * @param string $value User-selected option for default_reminder in agenda view
+	 * @return JSONResponse
+	 */
+	private function setDefaultReminder(string $value):JSONResponse {
+		if ($value !== 'none' &&
+			filter_var($value, FILTER_VALIDATE_INT,
+				['options' => ['max_range' => 0]]) === false) {
+			return new JSONResponse([], Http::STATUS_UNPROCESSABLE_ENTITY);
+		}
+
+		try {
+			$this->config->setUserValue(
+				$this->userId,
+				$this->appName,
+				'defaultReminder',
+				$value
+			);
+		} catch (\Exception $e) {
```

Comment:
this is not an expected exception, right?

Reply:
No, it just mirrors error handling as with all the other `set...` functions (e.g. `setSlotDuration`).

Looking at the manual, I see other examples from https://docs.nextcloud.com/server/latest/developer_manual/basics/storage/configuration.html don't do error handling, and https://github.com/nextcloud/server/blob/master/lib/public/IConfig.php doesn't mention any thrown exception from setUserValue(). 

I don't know if every similar try/catch should be dropped.

Reply:
ok, then let's keep it for consistency